lib/workers/repository/update/pr/changelog/release-notes.ts

```diff
@@ -166,10 +173,13 @@ export async function getReleaseNotes(
 
 function getExactReleaseMatch(
   packageName: string,
+  depName: string,
   version: string,
   releases: ChangeLogNotes[],
 ): ChangeLogNotes | undefined {
-  const exactReleaseReg = regEx(`${packageName}[@_-]v?${version}`);
+  const exactReleaseReg = regEx(
+    `(?:${depName}|${packageName})[@_-]v?${version}$`,
+  );
   const candidateReleases = releases.filter((r) => r.tag?.endsWith(version));
   const matchedRelease = candidateReleases.find((r) =>
     exactReleaseReg.test(r.tag!),
```

Comment:
Maybe slightly off-topic, but I wonder if the regex for `other` would match both `other-1.0.1` and `not-other-1.0.1`. 🤔

I'm tempted to prefix the regex with `(?:^|\/)` to avoid this kind of substring overmatching... maybe for another PR, but maybe worth doing here since `depName` is often shorter than `packageName` and could lead to more false partial matches.

WDYT?

Reply:
OK to change if existing tests pass